streams/src/main/java/org/apache/kafka/streams/processor/internals/StandbyTask.java

```diff
@@ -164,7 +165,29 @@ public void closeSuspended(final boolean clean,
     public List<ConsumerRecord<byte[], byte[]>> update(final TopicPartition partition,
                                                        final List<ConsumerRecord<byte[], byte[]>> records) {
         log.trace("Updating standby replicas of its state store for partition [{}]", partition);
-        return stateMgr.updateStandbyStates(partition, records);
+        final long limit = stateMgr.offsetLimit(partition);
+
+        final List<ConsumerRecord<byte[], byte[]>> toRestore = new ArrayList<>(records.size());
+        List<ConsumerRecord<byte[], byte[]>> remainingRecords = null;
+
+        long streamTime = -1L;
```

Comment:
Could we initialize streamTime as `((StandbyContextImpl) processorContext).streamTime()` instead? Otherwise in line 188 below we should only setStreamTime if the calculated `streamTime` is indeed larger, because if this fetched batch of records happen to have all timestamps smaller than the current stream time, then stream time will be set backwards.

Reply:
Hmm, true.

I don't think we should use just one "streamTime" for the whole context, since this StandbyTask bay have been assigned arbitrary changelog topics, so it would have a stream time unrelated to the StreamTasks that originally produced the results.

However, we could maintain a map of `TopicPartition -> streamTime` to prevent time from going backwards. Or we could just not worry about it. The only impact would be that some basically expired segments may be briefly resurrected before being cleaned up again later on.

Between the two options, I'm leaning more toward tracking the streamTime per topic-partition.